pandas/util/testing.py

```diff
@@ -97,11 +98,10 @@ def reset_display_options():
     """
     Reset the display options for printing and representing objects.
     """
-
     pd.reset_option("^display.", silent=True)
 
 
-def round_trip_pickle(obj, path=None):
+def round_trip_pickle(obj: FrameOrSeries, path: Optional[str] = None):
```

Comment:
```suggestion
def round_trip_pickle(obj: FrameOrSeries, path: Optional[str] = None) -> FrameOrSeries:
```

The power of the TypeVar here will come from using as a parameter annotation and as a return value type